src/components/captions-control/captions-control-mini.tsx

```diff
@@ -0,0 +1,112 @@
+import {createPortal} from 'preact/compat';
+import {registerToBottomBar} from '../bottom-bar';
+import {redux, ReservedPresetNames, style} from '../../index';
+import {Icon, IconType} from '../icon';
+import {h, VNode} from 'preact';
+import {ButtonControl} from '../button-control';
+import {Tooltip} from '../tooltip';
+import {Button} from '../button';
+import {CVAAOverlay} from '../cvaa-overlay';
+import {useState, useEffect} from 'preact/hooks';
+import {connect} from 'react-redux';
+import {Text} from 'preact-i18n';
+import {CaptionsMenu} from '../captions-menu';
+import {SmartContainer} from '../smart-container';
+import {CaptionsControl} from './captions-control';
+
+const mapStateToProps = state => ({
+  controlsToMove: state.bottomBar.controlsToMove
+});
+
+const COMPONENT_NAME = 'CaptionsControlMini';
```

Comment:
can you elaborate why do we need to duplicate the caption component

Reply:
since the `captionsControl` itself did not work as expected in the upper bar (while all the others are working just fine), the solution that was agreed is to add another "mini" component of it that would be rendered in the upper bar instead of the `captionsControl`, whenever it's needed.